### PR TITLE
fix(metadata-view): fix filtering folders and files

### PR DESCRIPTION
**Pull request by greg-in-a-box**
when selecting folders or files via the filter chip, they will be added to the query correctly now

## Summary by CodeRabbit

- New Features
  - Filtering now expands logical file-type selections into their mapped extensions and supports combining folder-type and file-type filters in a single query; introduces explicit file/folder type mappings to improve matching.

- Bug Fixes
  - Standardized mimetype-filter behavior for mixed folder/file selections and field names with special characters; unmapped values now yield no matches.

- Tests
  - Expanded coverage for mapped types, folder+file combos, indexing offsets, special-character field names, and edge cases.

### CODE REVIEW

File: src/elements/content-explorer/constants.ts [hunk 1]
```diff
@@ -2,12 +2,50 @@ import { FIELD_FILE_VERSION, FIELD_SHA1, FIELD_SHARED_LINK, FIELD_WATERMARK_INFO
...
+            'mts',
+            'qt',
+            'ts',
+            'flv',
+            'rm',
+        ],
+    ],
+    ['drawingType', ['dwg', 'dxf']],
+    ['threedType', ['obj', 'fbx', 'stl', 'amf', 'iges']],
+]);
```
[the remark this answers is not in the document]
Comment (tjiang-box) [reply src/elements/content-explorer/constants.ts]: @greg-in-a-box this is a valid comment. the duplicate line should be removed

**Review by tjiang-box — DISMISSED** [review]

**Review by jpan-box — DISMISSED** [review]

**Review by jpan-box — APPROVED** [review]
